code/controllers/subsystem/reagents.dm

```diff
@@ -0,0 +1,49 @@
+//Used for active reactions in reagents/equilibrium datums
+
+SUBSYSTEM_DEF(reagents)
+	name = "Reagents"
+	init_order = INIT_ORDER_REAGENTS
+	priority = FIRE_PRIORITY_REAGENTS
+	wait = 0.5 SECONDS //You might think that rate_up_lim has to be set to half, but since everything is normalised around delta_time, it automatically adjusts it to be per second. Magic!
+	flags = SS_KEEP_TIMING
+	runlevels = RUNLEVEL_GAME | RUNLEVEL_POSTGAME
+
+	var/list/processing = list()
+	var/list/currentrun = list()
+	///What time was it when we last ticked
+	var/previous_world_time = 0
+
+/datum/controller/subsystem/reagents/Initialize()
+	//So our first step isn't insane
+	previous_world_time = world.time
+	//Build GLOB lists - see holder.dm
+	build_chemical_reagent_list()
+	build_chemical_reactions_list()
+	return ..()
```

Comment:
Might be better to just override SSprocessing, since it does mostly the same stuff you are doing here.

Reply:
Having reagents process on it's own subsystem made the CPU useage efficiency increase by 4x. (at least according to the profiler 0.66 cpu useage across a reaction vs 0.33 with a 0.5 tick rate, I have the screenshots somewhere). Also live was having issues with slowdown during lag that was breaking reactions and getting people mad at me. SSprocessing also gives a delta_time independant of the difference between ticks - which I rely on in the code.

Reply:
I'm suggesting making this a processing subsystem, which in effect makes it a child of subsystem/processing, that way you can override the behavior you want to, I.E `fire()`, and keep the stuff you don't want to touch. look at the other subsystems in the processing folder, remember to use `PROCESSING_SUBSYSTEM_DEF()`